// Copyright 2011 Google Inc. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Note: Schema is subject to change. These objects should not be persisted.

syntax = "proto2";

package xpaf;

// A single output relation.
message Relation {
  optional string subject = 1;
  optional string predicate = 2;
  optional string object = 3;

  message Annotation {
    optional string name = 1;
    optional string value = 2;
  };
  repeated Annotation annotations = 4;

  // Copied directly from XpafParserDef.RelationTemplate.userdata.
  optional string userdata = 5;
};

// A single parser's output for a given document.
message ParserOutput {
  optional string parser_name = 1;
  repeated Relation relations = 2;
};

// Contains all ParserOutputs for a given document.
message ParsedDocument {
  optional string url = 1;
  repeated ParserOutput parser_outputs = 2;
};
